Source/FakeItEasy.Tests/Core/DynamicContainerTests.cs

```diff
@@ -106,22 +111,52 @@ public void Should_not_fail_when_more_than_one_configurator_exists_for_a_given_t
 
         private DynamicContainer CreateContainer()
         {
-            return new DynamicContainer(this.availableDummyFactories, this.availableConfigurators);
+            return new DynamicContainer(this.availableDummyFactories, this.availableOptionsBuilders);
         }
 
-        public class ConfigurationForTypeWithDummyFactory : FakeConfigurator<TypeWithDummyFactory>
+        public class OptionsBuilderForTypeWithDummyFactory : IFakeOptionsBuilder

```

Comment:
As it turns out, making the change does make the options builder simpler, but the test itself becomes more complicated:

```diff
         [Test]
         public void ConfigureFake_should_apply_configuration_from_registered_options_builders()
         {
             this.availableOptionsBuilders.Add(new OptionsBuilderForTypeWithDummyFactory());

             var container = this.CreateContainer();

-            var fakeOptions = A.Fake<IFakeOptions>();
+            var fakeOptions = A.Fake<IFakeOptions<TypeWithDummyFactory>>(options=>options.Implements<IFakeOptions>());
             var fakeTypeWithDummyFactory = A.Fake<TypeWithDummyFactory>();

-            A.CallTo(() => fakeOptions.ConfigureFake(A<Action<object>>._))
-                .Invokes((Action<object> action) => action(fakeTypeWithDummyFactory));
+            A.CallTo(() => fakeOptions.ConfigureFake(A<Action<TypeWithDummyFactory>>._))
+                .Invokes((Action<TypeWithDummyFactory> action) => action(fakeTypeWithDummyFactory));

-            container.BuildOptions(typeof(TypeWithDummyFactory), fakeOptions);
+            container.BuildOptions(typeof(TypeWithDummyFactory), (IFakeOptions) fakeOptions);

             Assert.That(fakeTypeWithDummyFactory.WasConfigured, Is.True, "configuration was not applied");
         }
```

This is because the `FakeOptionsBuilder<TFake>` has strict requirements about what time of options object is passed to it, and so on.
In my opinion, this makes the code more complicated than just implementing the `IFakeOptionsBuilder` by hand, but if you like it better, I'll go forward with the change.

Either way, `RobotRunsAmokEventFakeOptionsBuilder` can be converted easily. The other options builders all derive from `ConventionBasedOptionsBuilder` in `FakeOptionsBuilderSpecs`, and I think leaving them as is certainly saves code, but if you'd like to see them derive from `FakeOptionsBuilder<TFake>`, I can whip up a diff.